cli/src/commands/schema/edit.ts

```diff
@@ -294,593 +195,826 @@ Beware that this can lead to ${chalk.bold(
+        existingEntry.name = values.name;
+        existingEntry.nullable = parseBoolean(values.nullable) ?? true;
+        existingEntry.unique = parseBoolean(values.unique) ?? false;
+      } else if (!unchanged && !existingEntry) {
+        if (!this.columnEdits[column.tableName]) this.columnEdits[column.tableName] = {};
+        if (!this.columnEdits[column.tableName][column.originalName])
+          this.columnEdits[column.tableName][column.originalName] = {} as any;
+        this.columnEdits[column.tableName][column.originalName] = formatSchemaColumnToColumnData({
+          column: {
+            ...column,
+            ...values,
+            originalName: column.originalName,
+            notNull: parseBoolean(values.nullable) === false ? true : false,
+            unique: parseBoolean(values.unique) ? true : false
+          },
+          tableName: column.tableName
+        });
+      }
+      await this.showSchemaEdit();
+    } catch (err) {
+      if (err) throw err;
+    }
+  }
+
+  async showAddColumn({
+    tableName,
+    column
+  }: {
+    tableName: AddColumnPayload['tableName'];
+    column: AddColumnPayload['column'];
+  }) {
+    this.clear();
+    const template = `
+        name: \${name},
+        nullable: \${nullable},
+        unique: \${unique},
+        type: \${type},
+        default: \${default}
+        link: \${link}
+        vectorDimension: \${vectorDimension}
+        defaultPublicAccess: \${defaultPublicAccess}
+      },
+      table: ${tableName}`;
+
+    const snippet = new Snippet({
+      message: 'Add a column',
+      fields: [
+        {
+          name: 'name',
+          message: 'The name of the column',
+          validate: (value: string, state: ValidationState) => this.validateColumnName(value, state, column)
         },
         {
-          name: 'vectorDimension',
-          message: 'Vector Dimension. Only for columns that are vectors',
-          validate(
-            value: string | undefined,
-            state: ColumnEditState,
-            item: { value: string | undefined },
-            index: number
-          ) {
-            if (!isColumnAdded && value !== state.initial.vectorDimension) {
-              return `Cannot change the vector dimension of existing vector columns`;
-            }
-            if (state.values.type === 'vector') {
-              if (!value) {
-                return 'The vectorDimension field must be filled for columns of type `vector`';
-              }
-            } else if (value) {
-              return 'The vectorDimension field must not be filled unless the type of the column is `vector`';
-            }
-            return true;
+          name: 'type',
+          message: `The type of the column ${xataTypes}`,
+          validate: (value: string) => {
+            if (value === undefined) return 'Type cannot be undefined';
+            if (emptyString(value)) return 'Type cannot be empty';
+            if (!xataTypes.includes(value))
```

Comment:
Not allowing creation of types that are not "xata types"

Reply:
We should revisit this in a follow-up